src/cc/api/BPFTable.cc

```diff
@@ -142,26 +142,69 @@ StatusTuple BPFTable::remove_value(const std::string& key_str) {
   return StatusTuple(0);
 }
 
+StatusTuple BPFTable::clear_table_non_atomic() {
+  if (desc.type == BPF_MAP_TYPE_HASH ||
+      desc.type == BPF_MAP_TYPE_PERCPU_HASH ||
+      desc.type == BPF_MAP_TYPE_LRU_HASH ||
+      desc.type == BPF_MAP_TYPE_PERCPU_HASH ||
+      desc.type == BPF_MAP_TYPE_HASH_OF_MAPS) {
+    // For hash maps, use the first() interface (which uses get_next_key) to
+    // iterate through the map and clear elements
+    auto key = std::unique_ptr<void, decltype(::free)*>(
+      ::malloc(desc.key_size),
+      ::free);
+
+    while (this->first(key.get()))
+      if (!this->remove(key.get())) {
+        return StatusTuple(
+          -1,
+          "Failed to delete element when clearing table %s",
+          desc.name.c_str());
+      }
+  } else if (desc.type == BPF_MAP_TYPE_ARRAY ||
+             desc.type == BPF_MAP_TYPE_PERCPU_ARRAY) {
+    return StatusTuple(
+      -1, "Array map %s do not support clearing elements", desc.name.c_str());
+  } else if (desc.type == BPF_MAP_TYPE_PROG_ARRAY ||
+             desc.type == BPF_MAP_TYPE_PERF_EVENT_ARRAY ||
+             desc.type == BPF_MAP_TYPE_STACK_TRACE ||
+             desc.type == BPF_MAP_TYPE_ARRAY_OF_MAPS) {
+    // For Stack-trace and FD arrays, just iterate over all indices
+    for (size_t i = 0; i < desc.max_entries; i++)
+      if (!this->remove(&i)) {
+        return StatusTuple(
```

Comment:
When you deleted an element from an array, it is possible that that array index does not have a valid entry, so it will return an -ENOENT. One example is STACK_TRACE, not every array index will result in valid stack trace. Another is ARRAY_OF_MAPS, it may not be true every array index has corresponding maps.

So we should just call this->remove(&i) for all array elements and there is not need to care about the return value.